### PR TITLE
[Rm ChALL] Correct Figure captions for rst

### LINKED ISSUE #2875: [RM et al.] Fix figure captions
Here is a trivial example including an image.
For its part, numfig adds the numbering  (Fig. 15 VNF_cookbook in the following example)

```
.. figure:: ../figures/rc1_cookbook_vnf.png
   :alt: VNF_cookbook

   VNF_cookbook
```
They are many places across the streams where captions are falsy (+by hand)
Maybe it's due to the fact that the conversion from markdown was incorrect
_(I suggested to pick the flavor markdown and not the github one due to this particular issue)._

To illustrate the point, see RM chapter 7.
**Figure 7-1**: Reference Model Security Domains is even not considered as part of the image.

```
.. image:: ../figures/ch7_security_posture.png
   :alt: "Figure 7-1: Reference Model Security Domains"

**Figure 7-1**: Reference Model Security Domains
```
Here is the fix

 ```
.. image:: ../figures/ch7_security_posture.png
   :alt: "Reference Model Security Domains"

   Reference Model Security Domains
```

### PR DESCRIPTION
**Pull request by pgoyal01**
Partially Fixes #2875

### CODE REVIEW

**Comment by pgoyal01** [comment]
@cntt-n/ref-m This PR is now ready for review. Thanks

File: doc/ref_model/chapters/chapter07.rst [hunk 1]
```diff
@@ -140,10 +141,11 @@ The security certification of the platform will typically need to be the same, o
 
 The platform supports the workload, and in effect controls access to the workload from and to external endpoints such as carriage networks used by workloads, or by Data Centre Operations staff supporting the workload, or by tenants accessing workloads. From an access security perspective, the following diagram shows where different access controls will operate within the platform to provide access controls throughout the platform:
 
+.. :name: Reference Model Access Controls
 .. image:: ../figures/ch7-data-access-model.png
```
Comment (gkunz) [inline doc/ref_model/chapters/chapter07.rst:145]: Fixing linting error:
```
./chapters/chapter07.rst:145: D000 Error in "image" directive:
no content permitted.
```

```suggestion
.. figure:: ../figures/ch7-data-access-model.png
```

**Review by gkunz — CHANGES_REQUESTED** [review]
When building RM separately, the linter throws three errors. See the log of Cedric's automated check. I have indicated fixes for two.
The third error indicates that you need to rebase your PR because the current parent is _before_ the merged patch which fixes this error and _before_ the one which enables numfig.

**Review by gkunz — COMMENTED** [review]
LGTM, but there is one more instance of a non-numref reference to a figure in chapter3, line 274 (don't know why I cannot link to this)

**Review by gkunz — APPROVED** [review]
LGTM

File: doc/ref_model/chapters/chapter01.rst [hunk 2]
```diff
@@ -51,12 +51,13 @@ The document starts from the abstract and as it progresses it increasingly gets
 Scope
 -----
 
-This **Reference Model** document focuses on identifying the abstractions, and associated concepts, that are needed to represent the cloud infrastructure. **Figure 1-1** below highlights its scope in more details.
+This **Reference Model** document focuses on identifying the abstractions, and associated concepts, that are needed to represent the cloud infrastructure. :numref:`Fig-1-1` below highlights its scope in more details.
 
-.. image:: ../figures/ch01_scope.png
-   :alt: "Figure 1-1: Scope of Reference Model"
+.. _Fig-1-1:
```
Comment (karinesevilla) [inline doc/ref_model/chapters/chapter01.rst:56]: @pgoyal01 For the reference definition, I have a doubt about the writing and the underscore, would it not be .. _Fig-1-1 ?

Comment (karinesevilla) [reply doc/ref_model/chapters/chapter01.rst:56]: @pgoyal01 Agree with the format per documentation. I am seeing the text underlined, but it seems to be a bad effect of display. In your original file, it is correct, so do not take into account my comment.

File: doc/ref_model/chapters/chapter02.rst [hunk 3]
```diff
@@ -327,15 +327,16 @@ Profiles, Profile Extensions and Flavours will be considered in greater detail i
...
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Based on the above analysis, the following cloud infrastructure profiles are proposed (also shown in **Figure 2-5** below)
+Based on the above analysis, the following cloud infrastructure profiles are proposed (also shown in :numref:`Fig-2-5` below)
 
 -  **Basic**: for Workloads that can tolerate resource over-subscription and variable latency.
 -  **High Performance**: for Workloads that require predictable computing performance, high network throughput and low network latency.
 
-.. image:: ../figures/RM-ch02-node-profiles.png
+.. _Fig-2-5:
```
[the remark this answers is not in the document]
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter02.rst:335]: @pgoyal01 it is correct, so do not take into account my comment.

File: doc/ref_model/chapters/chapter05.rst [hunk 4]
```diff
@@ -6,15 +6,16 @@ A profile :ref:`ref_model/chapters/chapter02:profiles, profile extensions & flav
...
+Cloud Infrastructure Software layer is composed of 2 layers, :numref:`Cloud Infrastructure software layers`:
 
 -  The virtualisation Infrastructure layer, which is based on hypervisor virtualisation technology or container-based virtualisation technology. Container virtualisation can be nested in hypervisor-based virtualisation
 -  The host OS layer
 
-.. image:: ../figures/ch05-cloud-infrastructure-sw-profile-layers.png
-   :alt: "Figure 5-1: Cloud Infrastructure software layers"
+.. :name: Cloud Infrastructure software layers
+.. figure:: ../figures/ch05-cloud-infrastructure-sw-profile-layers.png
+   :alt: "FCloud Infrastructure software layers"
```
Comment (karinesevilla) [inline doc/ref_model/chapters/chapter05.rst:16]: @pgoyal01 typo: FCloud

File: doc/ref_model/chapters/chapter08.rst [hunk 5]
```diff
@@ -115,12 +118,13 @@ Introduction1
...
 
-As outlined in "Figure 8-3 - Conceptual Architecture of a Telco Operator Platform" above, the exposure point for the Management Interactions is the "Cloud Service Broker" and the "Cloud Resource Broker". The set of interactions that these interface points need to provide are defined by the "Figure 8-4 - Multi-Cloud Interaction Model" below. This provides a taxonomy for the interactions between the Communications Service Provider and the Cloud Providers.
+As outlined in :numref:`Conceptual Architecture of a Telco Operator Platform" above, the exposure point for the Management Interactions is the "Cloud Service Broker" and the "Cloud Resource Broker". The set of interactions that these interface points need to provide are defined by the :numref:`Multi-Cloud Interactions Model` below. This provides a taxonomy for the interactions between the Communications Service Provider and the Cloud Providers.
```
Comment (karinesevilla) [inline doc/ref_model/chapters/chapter08.rst:121]: @pgoyal01 Typo `Conceptual Architecture of a Telco Operator Platform"

**Review by karinesevilla — APPROVED** [review]

**Review by ulikleber — APPROVED** [review]

**Review by walterkozlowski — APPROVED** [review]

**Comment by collivier** [comment]
@walterkozlowski the gate voted -1 because they are issues in figure descriptions. I just fixed them.

**Pull request merged**